pkg/apis/serving/v1beta1/service_validation.go

```diff
@@ -53,11 +53,13 @@ func (s *Service) Validate(ctx context.Context) (errs *apis.FieldError) {
 // validateLabels function validates service labels
 func (s *Service) validateLabels() (errs *apis.FieldError) {
 	for key, val := range s.GetLabels() {
-		switch {
-		case key == serving.VisibilityLabelKey:
+		switch key {
+		case serving.VisibilityLabelKey:
 			errs = errs.Also(serving.ValidateClusterVisibilityLabel(val))
-		case strings.HasPrefix(key, serving.GroupNamePrefix):
-			errs = errs.Also(apis.ErrInvalidKeyName(key, apis.CurrentField))
+		default:
```

Comment:
The singleton cases feel like they should be `if/else` if we're not abusing `switch` as an `if/else` ladder anymore?

Reply:
I agree in this case (there are only two cases), but if you look at the more complex ones the switch seems to make sense as they grow. For example see the revision one.

Typically we are adding in cases for each specific known annotation to add in extra validation and the prefix is the catch-all if not known. In that sense the 'default:' structure of the switch flows pretty logically

Reply:
and the argument is to keep them all looking the same-ish so as we do add cases they keep consistent structure

Reply:
reverted for the two-case service validation